hathor/transaction/base_transaction.py

```diff
@@ -763,6 +795,24 @@ def start_mining(self, start: int = 0, end: int = MAX_NONCE, sleep_seconds: floa
                     return None
         return None
 
+    def set_height_hint(self, height: int) -> None:
+        """This method exists to set the height metadata when we can't calculate it yet (i.e. syncing checkpoints).
+
+        `TxValidationError` will be risen if a height already exists (which is not determined by its value but by the
+        validation state, that is, a height of `None` on a full validated transaction is final) and it doesn't match
+        the given hint. This is in place in order to prevent accidentally setting the height to a wrong value after the
+        value has been validated, the height must be checked when basic-validating this tx and effort should be made to
+        always set the correct height when using this method.
+        """
+        assert self.storage is not None
+        assert self.hash is not None
+        metadata = self.storage.get_metadata(self.hash)
+        if metadata is None:
+            metadata = TransactionMetadata(hash=self.hash, accumulated_weight=self.weight, height=height)
```

Comment:
> The only point I think we can do it safely is when syncing blocks between checkpoints from right to left. In this case, I feel this method should be placed in sync-v2 to avoid someone calling it from a different location.

That is exactly the only place where it is called from. What do you mean by placing it in sync-v2? Moving this change to another PR won't make any difference, or is it about moving the method? Since it touches "internals" (_metadata) I thought it was best to keep it on the `Transaction` class.

Reply:
You moved to the next PR. Great!